### ISSUE TITLE
Feature Request: Integration with Nextcloud

### ISSUE THREAD

**Issue by matbgn** [xemle/home-gallery#31]
Hi,

What an impressive job as a side project! Really.

Would you thing it's doable (time/cost reasonable) to link/point the storage with a Nextcloud instance? Taking advantage of NC's mobile apps, synchronization feature and so on, but with the power of Homegallery.

Do you already have a straightforward hack to do that (docker image bridge, script, etc.)?

What do you think of bridging Homegallery natively with Nextcloud as an integrated app (https://nextcloud.com/developer/)?

If neither of these approaches is to your liking, what solution would you propose to quickly and efficiently link the two applications?

**Comment by xemle** [comment]
Hi @matbgn 

> What an impressive job as a side project! Really.
Thank you for the flowers

> Would you thing it's doable (time/cost reasonable) to link/point the storage with a Nextcloud instance? Taking advantage of NC's mobile apps, synchronization feature and so on, but with the power of Homegallery.
> 
> Do you already have a straightforward hack to do that (docker image bridge, script, etc.)?
> 
> What do you think of bridging Homegallery natively with Nextcloud as an integrated app (https://nextcloud.com/developer/)?
> 
> If neither of these approaches is to your liking, what solution would you propose to quickly and efficiently link the two applications?

I do not use NextCloud. I am using [Syncthing](https://syncthing.net), a NAS and [Borg Backup](https://www.borgbackup.org) for my files. So my answer is only a best guess:

The easiest way: If you run docker (compose) you could setup NextCloud with HomeGallery sharing the data volume between both. This has the advantages that the files can be accessed directly and do not need the network (which comes with a high latency).

The more complex solution would be to implement a webdav feature for a file index and the extractor. Please have a look at the [indernals](https://docs.home-gallery.org/internals) to get a glimpse how the gallery works under the hood. Currently HomeGallery just reads media from a directory, creates the preview files in a storage directory and stores its configuration and databases in a configuration directory.

For NextCloud integration the question arise where to store the previews? In a separated directory or directly in NextCloud? The later would be great for public NextCloud services. However NextCloud is written in PHP while HomeGallery is written in NodeJS. So a container version would help here, too.

For the webdav integration I can give some support for someone but other features like automation and geo-map are currently on my top feature list. 

Maybe you can describe your usecase in detail for further discussions and solution proposals.

**Comment by matbgn** [comment]
Many thanks for your advices. I reserved some time on August to make the glue with Nextcloud trough a Docker. I come back to at this time.

**Comment by xemle** [comment]
@matbgn All right. See you then.

Please do not hesitate to ask further questions.

**Comment by tokenwizard** [comment]
For Nextcloud/Home-Gallery Shared functionality, I am just passing the Nextcloud storage location as a source in the Home-Gallery config file. 

FYI, I'm not running Nextcloud in Docker. Instead I'm using a turnkey Linux LXC template that included Nextcloud. 

in my case, the Nextcloud folder where my Phone snaps are automatically uploaded is: /var/www/nextcloud-data/james/files/Photos

So in my Home-Gallery config, I added that as a source for Photos:
![Selection_002](https://user-images.githubusercontent.com/24832561/211161170-7b3b62be-b437-46d1-be97-aa490ca4653b.png)

Now when I index the Home-Gallery sources, it pics up any newly-uploaded pictures I have snapped with my phone. As far as automatic indexing, I am currently just using a cron job that runs every 5 minutes and triggers the index/database update.
